backend/lucene/src/main/java/org/hibernate/search/backend/lucene/impl/LuceneBackendFactory.java

```diff
@@ -129,10 +129,27 @@ private Version getLuceneVersion(EventContext backendContext, ConfigurationPrope
 		return luceneVersion;
 	}
 
-	private MultiTenancyStrategy getMultiTenancyStrategy(ConfigurationPropertySource propertySource) {
-		MultiTenancyStrategyName multiTenancyStrategyName = MULTI_TENANCY_STRATEGY.get( propertySource );
+	private MultiTenancyStrategy getMultiTenancyStrategy(ConfigurationPropertySource propertySource,
+			EventContext eventContext, BackendBuildContext buildContext) {
+		Optional<MultiTenancyStrategyName> multiTenancyStrategyName = MULTI_TENANCY_STRATEGY.get( propertySource );
+		if ( !multiTenancyStrategyName.isPresent() ) {
+			// the default depends on mapping
+			return buildContext.multiTenancyEnabled() ?
+					new DiscriminatorMultiTenancyStrategy() : new NoMultiTenancyStrategy();
+		}
+
+		MultiTenancyStrategyName multiTenancyStrategy = multiTenancyStrategyName.get();
+		// check multiTenancyStrategy mismatch: required by the mapper vs explicitly configured with properties
+		if ( MultiTenancyStrategyName.NONE.equals( multiTenancyStrategy ) &&
+				buildContext.multiTenancyEnabled() ) {
+			throw log.multiTenancyRequiredButExplicitlyDisabledByBackend( eventContext );
+		}
+		if ( MultiTenancyStrategyName.DISCRIMINATOR.equals( multiTenancyStrategy ) &&
+				!buildContext.multiTenancyEnabled() ) {
+			throw log.multiTenancyNotRequiredButExplicitlyEnabledByTheBackend( eventContext );
+		}
```

Comment:
Same as Elasticsearch: you can get even better error messages, with the property name in the error message, by moving this check where you get the property value.